### PR TITLE
Find albs

### LINKED ISSUE #71: Identify ALBs to migrate
Find ALBs with DNS ready to be migrated

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- Find ALBs to migrate - fixes #71 
- Make `DomainRoute` and `CdnRoute` more compatible
- add function to instantiate correct `Migration` type

## Security considerations

None

### CODE REVIEW

File: migrator/migration.py [hunk 1]
```diff
@@ -13,24 +13,43 @@
...
 
 logger = logging.getLogger(__name__)
 
 
 def find_active_instances(session):
-    query = session.query(CdnRoute).filter(CdnRoute.state == "provisioned")
-    routes = query.all()
+    cdn_routes = find_active_cdn_instances(session)
+    domain_routes = find_active_domain_instances(session)
+    routes = [*cdn_routes, *domain_routes]
```
Comment (ccostino) [inline migrator/migration.py:25]: I initially had a question about the combination of these two route types here and if they'd be processed the same, but then I saw the factory below, so 👍 - nicely done!

File: migrator/migration.py [hunk 2]
```diff
@@ -13,24 +13,43 @@
...
+    return DomainMigration(route, session, client)
+
+
 def migrate_ready_instances(session, client):
     results = dict(migrated=[], skipped=[], failed=[])
     for route in find_active_instances(session):
         if route.has_valid_dns():
             try:
-                migration = Migration(route, session, client)
+                migration = migration_for_route(route, session, client)
```
Comment (ccostino) [inline migrator/migration.py:52]: Nice use of a factory!

File: migrator/migration.py [hunk 3]
```diff
@@ -402,3 +421,9 @@ def parse_cloudfront_error_response(error_responses):
         for item in error_responses.get("Items", []):
             responses[item["ResponseCode"]] = item["ResponsePagePath"]
         return responses
+
+
+class DomainMigration(Migration):
+    def __init__(self, route, session, client):
+        self.domains = route.domains
+        super().__init__(route, session, client)
```
Comment (ccostino) [inline migrator/migration.py:429]: Just checking that `DomainMigration` does not need the same attributes as the `CdnMigration` class, is that correct?

File: migrator/models.py [hunk 4]
```diff
@@ -90,14 +90,18 @@ class CdnCertificate(CdnBase):
 class DomainRoute(DomainBase):
     __tablename__ = "routes"
 
-    guid = sa.Column(sa.Text, primary_key=True)
+    instance_id = sa.Column("guid", sa.Text, primary_key=True)
```
Comment (ccostino) [inline migrator/models.py:93]: +1 for better readability of variables. :-)

**Review by ccostino — COMMENTED** [review]
Thanks, @bengerman13!  Just one question about the two classes (and a few callouts for A+ things).

File: migrator/migration.py (hunk 3 above)
Comment (bengerman13) [reply migrator/migration.py:429]: it doesn't need all of them - the broker's `ALBServiceInstance`s are way simpler than `CDNServiceInstance`s, so the `DomainMigration` is way simpler than the `CdnMigration`

**Review by ccostino — APPROVED** [review]
Thanks again, @bengerman13!

**Pull request merged**